CDCgov/prime-reportstream
# 8098 generate condition filtered FHIR bundle

This PR adds the ability to remove unfiltered conditions from a FHIR bundle. 

Test Steps:
1. Send the sample SR FHIR bundle attached using the reports API endpoint http://localhost:7071/api/reports with these  headers: client:ignore.ignore-fhir and Content-Type:application/fhir+ndjson
[sample_SR_bundle.txt](https://github.com/CDCgov/prime-reportstream/files/10886678/sample_SR_bundle.txt)
3. Inspect the generated HL7 file under your sftp folder and verify the file contains all observations
4. In organizations.yml, update the Ignore FULL_ELR receiver to have the following condition filter, replacing 600-7 with whatever loinc code shows up in your Postman test:
     `  conditionFilter: ["%obsPerformedCodes.intersect('600-7').exists()"]`
5. For your local env to pick up the changes that were made in this PR to the fhirpath_filter_shorthand.csv file you need to:
      ./gradlew resetDB
      ./gradlew run
      ./gradlew reloadTables
      To get your local env to pick up the changes that were made by you in your organizations.yml file you now need to run:
      ./gradlew reloadSettings
6. Re-send the sample SR FHIR bundle attached using the reports API
[sample_SR_bundle.txt](https://github.com/CDCgov/prime-reportstream/files/10886678/sample_SR_bundle.txt)
7. Compare the two generated HL7 files under your sftp folder and verify the new file only contains OBX segments that have loinc codes of 600-7 and are not AOE questions. `Ex: OBX|1|CE|600-7^COVID-19^LN||260373001^POSITIVE^SCT||||||F||||||||||||Third^^^^^^^^^12D3456789|1 Main St^^^IG^99551^USA`

## Changes
- Added function that removes Observations from a bundle that are not listed under an Endpoint resource
- Added unit tests

## Checklist

### Testing

## Linked Issues
- Fixes #8098
